templates/pkg/crd_sdk.go.tpl

```diff
@@ -40,13 +43,34 @@ func (rm *resourceManager) sdkFind(
 	if err != nil {
 		return nil, err
 	}
+	//If the input implements validate method, invoke Validate() before making ReadOne call
+	inputType := reflect.TypeOf(input)
+	_, validateable := inputType.MethodByName("Validate")
+	if validateable {
+		err := input.Validate()
```

Comment:
Maybe we can just use interface type assertions here instead of using the `reflect` package ?

Reply:
I think it is not possible to directly check (on runtime) if a type implement an interface. But with interface conversation we can achieve that with something like this:

```golang
package main

import "fmt"

type Validatable interface {
	Valid() error
}

type Input struct{}

func (i *Input) Valid() error { return nil }

func isValidatable(t interface{}) bool {
	_, ok := t.(Validatable)
	return ok
}

func main() {
	i := &Input{}
	fmt.Println(isValidatable(i))
}

```